src/index.ts

```diff
@@ -1,3 +1,64 @@
-export default function greeter(name: string): string {
-  return `Hello, ${name}!`;
-}
+import { ethErrors, serializeError } from 'eth-rpc-errors';
+import openrpcDocument from './openrpc.json';
+import methodMappingFactory from './methods/methodMapping';
+import IframeExecutionEnvironmentTransport from './transports/IframeExecutionEnvironmentTransport';
+
+const methodMapping = methodMappingFactory(
```

Comment:
Nit: Well, now this is a singleton :sweat_smile: 

Granted it's less weird to have a singleton in the entrypoint for the web app, but it'll still present the same challenges for testing the message event handler (no easy way to stub out the method mapping). We could for example create a class that created this in its constructor. Then we could monkey patch the method mapping to test the message handler easily enough.

We'd still need to construct this class directly in module scope to start things running, but at least the bulk of the logic would be easily testable then. The entrypoint is always near impossible to unit test thoroughly.

Reply:
Resolved - the latest commit addresses this